protocols/kad/src/behaviour.rs

```diff
@@ -2539,18 +2563,8 @@ pub type GetRecordResult = Result<GetRecordOk, GetRecordError>;
 /// The successful result of [`Kademlia::get_record`].
 #[derive(Debug, Clone)]
 pub struct GetRecordOk {
-    /// The records found, including the peer that returned them.
-    pub records: Vec<PeerRecord>,
-    /// If caching is enabled, these are the peers closest
-    /// _to the record key_ (not the local node) that were queried but
-    /// did not return the record, sorted by distance to the record key
-    /// from closest to farthest. How many of these are tracked is configured
-    /// by [`KademliaConfig::set_caching`]. If the lookup used a quorum of
-    /// 1, these peers will be sent the record as a means of caching.
-    /// If the lookup used a quorum > 1, you may wish to use these
-    /// candidates with [`Kademlia::put_record_to`] after selecting
-    /// one of the returned records.
-    pub cache_candidates: BTreeMap<kbucket::Distance, PeerId>,
```

Comment:
I think there was a misunderstanding here.

> This pull request removes the quorum mechanism. To keep things simple, I suggest also removing the automated replication to the closest node that does not have the value, but instead entirely rely on the user to call `put_record_to`.

`https://github.com/libp2p/rust-libp2p/pull/2712/#discussion_r984483930`

You removed the automated caching in `query_finished`. That is good. We now rely on the user to call `put_record_to`. But given that you remove the `cache_candidates` on the `GetRecordOk` `struct`, the user no longer knows with which peers it is supposed to call `put_record_to`. We still need to track and emit `cache_candidates`.

Does the above make sense?

Reply:
Yes very much, my mistake entirely. I brought back `KademliaCaching` and `cache_candidates` to enable tracking this now.